flytekit/core/type_engine.py

```diff
@@ -1090,6 +1090,16 @@ def get_sub_type(t: Type[T]) -> Type[T]:
 
         raise ValueError("Only generic univariate typing.List[T] type is supported.")
 
+    @staticmethod
+    def get_sub_type_or_none(t: Type[T]) -> Optional[Type[T]]:
```

Comment:
I chose to add a new method because `get_sub_type` is used in many places where they assume it never returns `None`.